docs/source/getting_started/benchmarks/cifar10_benchmark.py

```diff
@@ -413,6 +421,25 @@ def configure_optimizers(self):
         return [optim], [scheduler]
 
 
+class FastSiamModel(SimSiamModel):
+    def __init__(self, dataloader_kNN, num_classes):
+        super().__init__(dataloader_kNN, num_classes)
+
+    def training_step(self, batch, batch_idx):
+        xs, _, _ = batch
+        zs, ps = zip(*list(self.forward(x) for x in xs))
+        zs = torch.cat([z.unsqueeze(0) for z in zs], dim=0)
+        ps = torch.cat([p.unsqueeze(0) for p in ps], dim=0)
+
+        loss = 0.0
+        for i in range(len(xs)):
+            mask = torch.arange(len(xs)) != i
+            loss += self.criterion(ps[i], torch.mean(zs[mask], dim=0))
```

Comment:
I am surprised that you don't have to specify the device for the mask. If you want to add it you can do:
```
mask = torch.arange(len(xs), device=self.device) != i
```